operator/v1/types.go

```diff
@@ -258,15 +258,21 @@ type StaticPodOperatorStatus struct {
 
 // NodeStatus provides information about the current state of a particular node managed by this operator.
 // +kubebuilder:validation:XValidation:rule="has(self.currentRevision) || !has(oldSelf.currentRevision)",message="cannot be unset once set",fieldPath=".currentRevision"
+// +kubebuilder:validation:XValidation:rule="oldSelf.hasValue() || (has(self.currentRevision) ? self.currentRevision == 0 : true) ",message="when specified on creation of a nodeStatus, currentRevision must be set to 0 on creation",optionalOldSelf=true
+// +kubebuilder:validation:XValidation:rule="oldSelf.hasValue() || (has(self.targetRevision) ? self.targetRevision == 0 : true) ",message="when specified on creation of a nodeStatus, targetRevision must be set to 0 on creation",optionalOldSelf=true
 type NodeStatus struct {
 	// nodeName is the name of the node
 	// +required
 	NodeName string `json:"nodeName"`
 
-	// currentRevision is the generation of the most recently successful deployment
+	// currentRevision is the generation of the most recently successful deployment.
+	// If set on creation of a nodeStatus, it must be set to 0. Updates must only increase the value.
 	// +kubebuilder:validation:XValidation:rule="self >= oldSelf",message="must only increase"
+	// +optional
```

Comment:
Package level optional setting has saved you here, this should have been required but the package defaults to optional.

In which case, this should probably have omitempty as clients should already expect to not have a value here.

Is there any valid reason for a go client to set this value to 0?

Reply:
Happy to add the omitempty.

I am not aware of any valid reason for a go client to set this value to 0.

Reply:
Added omitempty in the latest commit.